### PR TITLE
Removing optionparser in favor of Thor

### PR DESCRIPTION
**Pull request by danzilio**

### CODE REVIEW

File: lib/modulesync/cli.rb [hunk 1]
```diff
@@ -1,121 +1,47 @@
...
       end
 
-      @options.fetch(:command) do
-        fail('A command is required.')
-      end
-
-      if @options[:command] == 'hook' &&
-         (!args.include?('activate') && !args.include?('deactivate'))
-        fail('You must activate or deactivate the hook.')
+      desc 'update', 'Activat or deactivate a git hook.'
```
Comment (blkperl) [inline lib/modulesync/cli.rb]: typo of activate

File: lib/modulesync/cli.rb [hunk 2]
```diff
@@ -1,121 +1,47 @@
...
+        ModuleSync.run(config)
       end
 
-      @options.fetch(:command) do
-        fail('A command is required.')
-      end
-
-      if @options[:command] == 'hook' &&
-         (!args.include?('activate') && !args.include?('deactivate'))
-        fail('You must activate or deactivate the hook.')
```
Comment (blkperl) [inline lib/modulesync/cli.rb]: use ` fail Thor::Error, msg`

File: lib/modulesync/cli.rb [hunk 3]
```diff
@@ -1,121 +1,47 @@
...
+      option :changelog, :type => :boolean, :desc => 'Update CHANGELOG.md if version was bumped', :default => false
+      option :tag, :type => :boolean, :desc => 'Git tag with the current module version', :defalut => false
+      option :tag_pattern, :desc => 'The pattern to use when tagging releases.'
+
+      def update
+        config = {:command => 'update'}.merge(options)
+        config.merge!(Util.parse_config(MODULESYNC_CONF_FILE))
+        config = config.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}
+        puts config
+        fail 'No value provided for required option "--message"' unless config[:noop] || config[:message] || config[:offline]
```
Comment (blkperl) [inline lib/modulesync/cli.rb]: use ` fail Thor::Error, msg`

**Comment by danzilio** [comment]
Not entirely sure how confident I am in the tests though...

**Comment by danzilio** [comment]
This should be merged after #61

**Comment by danzilio** [comment]
This will close #21

**Pull request merged**